### PR TITLE
[kube-prometheus-stack] Add IsDefaultDatasource for editing isDefault value for another DS

### LINKED ISSUE #482: [prometheus-kube-stack] grafana CrashLoopBackOff with `additionalDataSources`
**Describe the bug**
Grafana becomes `CrashLoopBackOff` with `additionalDataSources`.

`pod/kube-prometheus-stack-grafana-96bcc44f-hx9nw` is `CrashLoopBackOff` below.

```shell
➜  capa-viewer git:(main) ✗ (☸ |kind-kind:ops) kubectl get all                                                   
NAME                                                            READY   STATUS             RESTARTS   AGE
pod/kube-prometheus-stack-grafana-96bcc44f-hx9nw                1/2     CrashLoopBackOff   6          8m19s
pod/kube-prometheus-stack-kube-state-metrics-5cf575d8f8-6525z   1/1     Running            0          8m19s
pod/kube-prometheus-stack-operator-5d77bf74-d2jw9               1/1     Running            0          8m19s
pod/kube-prometheus-stack-prometheus-node-exporter-5jzcl        1/1     Running            0          8m19s
pod/kubernetes-dashboard-65585fcf98-z6bs8                       1/1     Running            0          8m29s
pod/prometheus-kube-prometheus-stack-prometheus-0               2/2     Running            1          8m16s

NAME                                                     TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
service/kube-prometheus-stack-alertmanager               ClusterIP   10.96.170.88    <none>        9093/TCP   8m19s
service/kube-prometheus-stack-grafana                    ClusterIP   10.96.249.168   <none>        80/TCP     8m19s
service/kube-prometheus-stack-kube-state-metrics         ClusterIP   10.96.198.236   <none>        8080/TCP   8m19s
service/kube-prometheus-stack-operator                   ClusterIP   10.96.17.128    <none>        443/TCP    8m19s
service/kube-prometheus-stack-prometheus                 ClusterIP   10.96.17.252    <none>        9090/TCP   8m19s
service/kube-prometheus-stack-prometheus-node-exporter   ClusterIP   10.96.254.179   <none>        9100/TCP   8m19s
service/kubernetes-dashboard                             ClusterIP   10.96.37.110    <none>        443/TCP    8m29s
service/prometheus-operated                              ClusterIP   None            <none>        9090/TCP   8m16s

NAME                                                            DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
daemonset.apps/kube-prometheus-stack-prometheus-node-exporter   1         1         1       1            1           <none>          8m19s

NAME                                                       READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/kube-prometheus-stack-grafana              0/1     1            0           8m19s
deployment.apps/kube-prometheus-stack-kube-state-metrics   1/1     1            1           8m19s
deployment.apps/kube-prometheus-stack-operator             1/1     1            1           8m19s
deployment.apps/kubernetes-dashboard                       1/1     1            1           8m29s

NAME                                                                  DESIRED   CURRENT   READY   AGE
replicaset.apps/kube-prometheus-stack-grafana-96bcc44f                1         1         0       8m19s
replicaset.apps/kube-prometheus-stack-kube-state-metrics-5cf575d8f8   1         1         1       8m19s
replicaset.apps/kube-prometheus-stack-operator-5d77bf74               1         1         1       8m19s
replicaset.apps/kubernetes-dashboard-65585fcf98                       1         1         1       8m29s

NAME                                                           READY   AGE
statefulset.apps/prometheus-kube-prometheus-stack-prometheus   1/1     8m16s
```

```shell
➜  capa-viewer git:(main) ✗ (☸ |kind-kind:ops) kubectl logs kube-prometheus-stack-grafana-96bcc44f-hx9nw grafana 
t=2020-12-11T08:26:34+0000 lvl=info msg="Starting Grafana" logger=server version=7.2.1 commit=72a6c64532 branch=HEAD compiled=2020-10-08T09:00:32+0000
t=2020-12-11T08:26:34+0000 lvl=info msg="Config loaded from" logger=settings file=/usr/share/grafana/conf/defaults.ini
t=2020-12-11T08:26:34+0000 lvl=info msg="Config loaded from" logger=settings file=/etc/grafana/grafana.ini
t=2020-12-11T08:26:34+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.data=/var/lib/grafana"
t=2020-12-11T08:26:34+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.logs=/var/log/grafana"
t=2020-12-11T08:26:34+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.plugins=/var/lib/grafana/plugins"
t=2020-12-11T08:26:34+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.provisioning=/etc/grafana/provisioning"
t=2020-12-11T08:26:34+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.log.mode=console"
t=2020-12-11T08:26:34+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_PATHS_DATA=/var/lib/grafana"
t=2020-12-11T08:26:34+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_PATHS_LOGS=/var/log/grafana"
t=2020-12-11T08:26:34+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_PATHS_PLUGINS=/var/lib/grafana/plugins"
t=2020-12-11T08:26:34+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_PATHS_PROVISIONING=/etc/grafana/provisioning"
t=2020-12-11T08:26:34+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_SECURITY_ADMIN_USER=foo"
t=2020-12-11T08:26:34+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_SECURITY_ADMIN_PASSWORD=*********"
t=2020-12-11T08:26:34+0000 lvl=info msg="Path Home" logger=settings path=/usr/share/grafana
t=2020-12-11T08:26:34+0000 lvl=info msg="Path Data" logger=settings path=/var/lib/grafana
t=2020-12-11T08:26:34+0000 lvl=info msg="Path Logs" logger=settings path=/var/log/grafana
t=2020-12-11T08:26:34+0000 lvl=info msg="Path Plugins" logger=settings path=/var/lib/grafana/plugins
t=2020-12-11T08:26:34+0000 lvl=info msg="Path Provisioning" logger=settings path=/etc/grafana/provisioning
t=2020-12-11T08:26:34+0000 lvl=info msg="App mode production" logger=settings
t=2020-12-11T08:26:34+0000 lvl=info msg="Connecting to DB" logger=sqlstore dbtype=sqlite3
t=2020-12-11T08:26:34+0000 lvl=info msg="Starting DB migrations" logger=migrator
t=2020-12-11T08:26:34+0000 lvl=info msg="Starting plugin search" logger=plugins
t=2020-12-11T08:26:34+0000 lvl=info msg="Registering plugin" logger=plugins name="Direct Input"
t=2020-12-11T08:26:34+0000 lvl=eror msg="Server shutdown" logger=server reason="Service init failed: Datasource provisioning error: datasource.yaml config is invalid. Only one datasource per organization can be marked as default"
```

**Version of Helm and Kubernetes**:

Helm Version:

```console
$ helm version
version.BuildInfo{Version:"v3.4.1", GitCommit:"c4e74854886b2efe3321e185578e6db9be0a6e29", GitTreeState:"clean", GoVersion:"go1.14.11"}
```

Kubernetes Version:

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"clean", BuildDate:"2020-11-16T17:18:52Z", GoVersion:"go1.15.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.1", GitCommit:"206bcadf021e76c27513500ca24182692aabd17e", GitTreeState:"clean", BuildDate:"2020-09-14T07:30:52Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}
```

**Which chart**: prometheus-kube-stack

**Which version of the chart**: 12.7.0

**What happened**:
Grafana becomes `CrashLoopBackOff` with `additionalDataSources`.

**What you expected to happen**:
It should 'just work'.

**How to reproduce it** (as minimally and precisely as possible):
**Changed values of values.yaml** (only put values which differ from the defaults):

values.yaml

```yaml
grafana:
  adminUser: foo
  adminPassword: bar
  persistence:
    enabled: true
  additionalDataSources:
    - name: Loki
      type: loki
      url: http://loki-stack:3100
      access: proxy
      isDefault: true
      version: 1
```

**The helm command that you execute and failing/misfunctioning**:

For example:

```console
helm install kube-prometheus-stack prometheus-community/kube-prometheus-stack --values values.yaml
```

**Helm values set after installation/upgrade:**

```yaml
grafana:
  additionalDataSources:
  - access: proxy
    isDefault: true
    name: Loki
    type: loki
    url: http://loki-stack:3100
    version: 1
  adminPassword: bar
  adminUser: foo
  enabled: true
  persistence:
    enabled: true
```

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by MaxMuzh**
Signed-off-by: Max Muzh <mmuzh@augury.com>

#### What this PR does / why we need it
- Right now we can't add another default Prometheus DataSource without deleting the default Prometheus that we wish to keep even if it's not as a default Prometheus DS.

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes the addition of Thanos as default datasource without deleting the Prometheus default datasource.
Unfortunately when I add `grafana.additionalDataSources` with `isDefault` that is `true`, it won't create it because of collision (Prometheus default datasource)
- Also fixes `fixes #482` and and give the option to avoid `#2251` (both closed but not solved)

#### Special notes for your reviewer

- Please be gentle because this is my first PR here

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/values.yaml [hunk 1]
```diff
@@ -793,6 +793,7 @@ grafana:
     datasources:
       enabled: true
       defaultDatasourceEnabled: true
+      IsDefaultDatasource: true
```
Comment (MaxMuzh) [inline charts/kube-prometheus-stack/values.yaml:796]: I'm totally open for a variable name suggestions

Comment (QuentinBisson) [reply charts/kube-prometheus-stack/values.yaml:796]: I would probably suggest isDefault here

**Comment by QuentinBisson** [comment]
Can you fix the conflicts?

File: charts/kube-prometheus-stack/values.yaml (hunk 1 above)
Comment (monotek) [inline charts/kube-prometheus-stack/values.yaml:796]: Variable should start with lowercase letter.

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by MaxMuzh** [comment]
@monotek @QuentinBisson  Done

**Review by monotek — APPROVED** [review]

**Pull request merged**